react/components/RenderProvider.tsx

```diff
@@ -343,9 +354,15 @@ class RenderProvider extends Component<Props, RenderProviderState> {
   }
 
   public UNSAFE_componentWillReceiveProps(nextProps: Props) {
-    // If RenderProvider is being re-rendered, the global runtime might have changed
-    // so we must update all extensions.
-    if (this.rendered) {
+    const updatedProps = difference(
```

Comment:
Why do a whole diff, and not only compare nextProps.runtime !== this.props.runtime?

It is usually not a good idea to do such a diff on cWRP, but you already know that

Reply:
i did this because i think it is safer, and cWRP on this component is seldom called, and the diff is not that costly anyway

I'll look into it tho

Reply:
I've measured it, it takes about 1~3ms, and rarely runs.